src/lib/sw_engine/tvgSwRaster.cpp

```diff
@@ -133,6 +134,7 @@ static bool _rasterSolidRle(SwSurface* surface, SwRleData* rle, uint32_t color)
 static bool _rasterLinearGradientRect(SwSurface* surface, const SwBBox& region, const SwFill* fill)
 {
     if (!fill) return false;
+    if (fill->linear.len < FLT_EPSILON) return false;
```

Comment:
Rather than adding exception all cases, you can skip drawing in this entry point one time: rasterGradientShape()

Reply:
I know I can but then I have to check if shape->fill exists there, then if it's linear or radial, and then do correct check. I though it makes more sense to put the check in place where we know what kind of fill we have